include/ginkgo/core/base/lin_op.hpp

```diff
@@ -791,13 +791,19 @@ public:                                                                      \
 // A workaround for the NVCC compiler - parameter pack expansion does not work
 // properly. You won't be able to use factories in code compiled with NVCC, but
 // at least this won't trigger a compiler error as soon as a header using it is
-// included.
-#define GKO_FACTORY_PARAMETER(_name, ...) \
-    mutable _name{__VA_ARGS__};           \
-                                          \
-    template <typename... Args>           \
-    auto with_##_name(Args &&... _value)  \
-        const->const ::gko::xstd::decay_t<decltype(*this)> &
+// included. To not get a linker error, we provide a dummy body.
+#define GKO_FACTORY_PARAMETER(_name, ...)                                    \
```

Comment:
So, this works for the `hcc` compiler on AMD ? If not then you maybe have to add `__HIPCC__` ?

Reply:
Good point again, I will have to test this. So far, no problems were seen, but I am unsure if that's because there actually is no problem or because we anyway do not use this.

Reply:
I am not pretty sure whether the following thing is related to here.
When I try to compile the solver test with hpicc, the criterion settings are always failed.
The criterion do not changed by with_iteration etc, so the criterion is always nullptr such the solver can not work,
However, I use gcc to compile the test successfully.

Reply:
I will try to add `__HIPCC__` here. I cannot test it because I do not have any solver implemented in my PR. If it does not work in your PR, we will have to think of something else @yhmtsai.